ros2doctor/test/test_report_environment.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2025 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.actions import SetEnvironmentVariable
+from launch.actions import UnsetEnvironmentVariable
+
+import launch_testing
+import launch_testing.actions
+import launch_testing.markers
+
+import pytest
+
+from rclpy.utilities import get_available_rmw_implementations
+from ros2doctor.api import Report
+from ros2doctor.api.environment import EnvironmentReport
+
+
+@pytest.mark.rostest
+@launch_testing.parametrize('rmw_implementation', get_available_rmw_implementations())
+@launch_testing.markers.keep_alive
+def generate_test_description(rmw_implementation: str):
+    return LaunchDescription([
+        # Always restart daemon to isolate tests.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=[
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        UnsetEnvironmentVariable('ROS_AUTOMATIC_DISCOVERY_RANGE'),
+                        UnsetEnvironmentVariable('ROS_DISTRO'),
```

Comment:
(just a comment to make sure i understand the intention here) These are set by default and can be dependent on distribution and default value, so we should exclude those environment variables for test.